source/common/router/router.cc

```diff
@@ -1510,6 +1454,95 @@ bool Filter::setupRedirect(const Http::ResponseHeaderMap& headers,
   return false;
 }
 
+bool Filter::convertRequestHeadersForInternalRedirect(Http::RequestHeaderMap& downstream_headers,
+                                                      const Http::HeaderEntry& internal_redirect) {
+  if (!downstream_headers.Path()) {
+    ENVOY_STREAM_LOG(trace, "no path in downstream_headers", *callbacks_);
+    return false;
+  }
+
+  // Make sure the redirect response contains a URL to redirect to.
+  if (internal_redirect.value().getStringView().length() == 0) {
+    config_.stats_.passthrough_internal_redirect_bad_location_.inc();
+    return false;
+  }
+  Http::Utility::Url absolute_url;
+  if (!absolute_url.initialize(internal_redirect.value().getStringView(), false)) {
+    config_.stats_.passthrough_internal_redirect_bad_location_.inc();
+    return false;
+  }
+
+  auto& policy = route_entry_->internalRedirectPolicy();
```

Comment:
not your code, but const all variables that can be const?

Reply:
Done.